@here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts

```diff
@@ -1805,6 +1776,207 @@ export class OmvDecodedTileEmitter implements IOmvEmitter {
+            const line: number[] = [];
+            const lineUvs: number[] = [];
+            const lineOffsets: number[] = [];
+            const untiledLine = this.computeUntiledLine(polyline);
+
+            const pointsInside = positions.map(
+                pos => pos.x < extents && pos.x > 0.0 && pos.y < extents && pos.y > 0.0
+            );
+
+            const iterations = positions.length - 1;
+            for (let i = 0; i < iterations; i++) {
+                const currentPosIntersects = pointsInside[i];
+                const nextPosPosIntersects = pointsInside[i + 1];
+                const currentPos = positions[i];
+                const nextPos = positions[i + 1];
+
+                if (currentPosIntersects === false && nextPosPosIntersects === false) {
+                    intersect(line, lineUvs, lineOffsets, untiledLine, currentPos, nextPos);
+                } else if (currentPosIntersects === true && nextPosPosIntersects === true) {
+                    if (i === 0) {
+                        this.addPoint(
+                            currentPos,
+                            line,
+                            lineUvs,
+                            lineOffsets,
+                            untiledLine,
+                            extents,
+                            computeTexCoords
+                        );
+                    }
+                    this.addPoint(
+                        nextPos,
+                        line,
+                        lineUvs,
+                        lineOffsets,
+                        untiledLine,
+                        extents,
+                        computeTexCoords
+                    );
+                } else if (currentPosIntersects !== nextPosPosIntersects) {
+                    if (i === 0 && currentPosIntersects) {
+                        this.addPoint(
+                            currentPos,
+                            line,
+                            lineUvs,
+                            lineOffsets,
+                            untiledLine,
+                            extents,
+                            computeTexCoords
+                        );
+                    }
+                    intersect(line, lineUvs, lineOffsets, untiledLine, currentPos, nextPos);
+                    if (nextPosPosIntersects) {
+                        this.addPoint(
+                            nextPos,
+                            line,
+                            lineUvs,
+                            lineOffsets,
+                            untiledLine,
+                            extents,
+                            computeTexCoords
+                        );
+                    }
+                } else {
+                    throw new Error("NEVER");
+                }
+            }
+            lines.push(line);
+            uvs.push(lineUvs);
+            offset.push(lineOffsets);
+        });
+    }
+
+    private addPoint(
```

Comment:
``addPoint`` is a bit generic, since we also work with point features in this file. Please rename to something along the lines of ``addLinePoint`` or something like that.